monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -875,7 +875,29 @@ object Task extends TaskInstances {
     */
   def gather[A, M[X] <: TraversableOnce[X]](in: M[Task[A]])
     (implicit cbf: CanBuildFrom[M[Task[A]], A, M[A]]): Task[M[A]] =
-    TaskGather(in)(cbf)
+    TaskGather(in.toArray, () => cbf(in))
```

Comment:
I changed `TaskGather` to take a `TraversableOnce[Task[A]]` parameter - this helped address the next issue as well.

(the `.toArray` being there is an artifact of a previous draft, when the code still took in `CanBuildFrom` instead of a builder maker)